### ISSUE TITLE
Outer boundary tail is dropped for modes with negative rho_lm at r_max

### ISSUE THREAD

**Issue by nstarman** [jnibauer/bfeax#1]
## Summary

The outer boundary tail correction in the radial Poisson solve is gated on the
*signed* value of `rho_lm` at `r_max`, so any mode whose coefficient is
negative there has its tail correction silently zeroed instead of applied with
the correct sign. For `l >= 1` the `rho_lm` coefficients are routinely
negative, so this affects a large fraction of the anisotropic modes.

Found while porting this machinery into
[`galax`](https://github.com/GalacticDynamics/galax). Reported at commit
`2341204`.

## Location

`bfeax/potential.py:137` (the live `_poisson_scan` path), and identically
`bfeax/poisson.py:107`:

```python
active_out = rho_col[-1] > 0.0
...
dI_out = jnp.where(active_out & tail_conv, rho_col[-1] * ... , 0.0)
```

The `dI_out` expression itself is already correctly signed — it uses
`rho_col[-1]`, not `|rho_col[-1]|`. Only the *gate* rejects negative modes.

Note the asymmetry with the inner tail a few lines above, which does handle
sign correctly:

```python
A_in      = jnp.sign(rho_col[0]) * jnp.abs(rho_col[0]) * ...   # signed
active_in = jnp.abs(rho_col[0]) > 1e-8 * rho_sc                # sign-agnostic gate
```

## Reproduction

Triaxial NFW, `p=0.8`, `q=0.5`, `r_min=1e-2`, `r_max=300`, `n_r=128`,
`l_max=8`, `symmetry="triaxial"`. Of the 15 retained modes, 6 have
`rho_lm(r_max) < 0` and lose their tail:

```
mode     rho_lm(r_max)    tail applied?
(0, 0)    +5.224e-08      yes
(2, 0)    -2.430e-08      no     <-- dominant anisotropic mode
(2, 2)    +1.016e-08      yes
(4, 0)    +8.718e-09      yes
(4, 2)    -4.185e-09      no
(4, 4)    +1.268e-09      yes
(6, 0)    -2.937e-09      no
(6, 2)    +1.690e-09      yes
(6, 4)    -4.980e-10      no
(6, 6)    +1.520e-10      yes
(8, 0)    +8.830e-10      yes
(8, 2)    -5.900e-10      no
(8, 4)    +1.965e-10      yes
(8, 6)    -5.694e-11      no
(8, 8)    +1.779e-11      yes
```

## Impact

Changing *only* the gate to be sign-agnostic (`jnp.abs(rho_col[-1]) > 0.0`)
leaves the 9 positive modes bit-identical and changes the 6 negative ones by
40-47% in `Phi_lm` near `r_max`:

```
mode     max |dPhi_lm / Phi_lm|   at r
(2, 0)              40.21%        300
(4, 2)              44.48%        300
(6, 0)              45.96%        300
(6, 4)              45.97%        300
(8, 2)              46.56%        300
(8, 6)              46.57%        300
```

Summed into the full potential the effect is naturally much smaller, since the
monopole dominates and is unaffected — but it grows toward `r_max`:

```
point (kpc)              Phi rel err    |accel| rel err
(5, 3, 2)                     0.000%             0.000%
(20, 10, 5)                   0.000%             0.001%
(60, 40, 30)                  0.007%             0.040%
(150, 80, 60)                 0.090%             0.391%
(250, 150, 100)               0.418%             1.709%
```

So: negligible in the inner halo, but ~1.7% in acceleration at `r = 250` with
`r_max = 300`. That is above the ~0.1% accuracy the README targets for
`n_r=128`, and it is concentrated exactly where the README notes the outer tail
integral is critical for NFW.

## Suggested fix

Make the gate sign-agnostic while leaving the (already signed) `dI_out`
expression alone:

```python
active_out = jnp.abs(rho_col[-1]) > 0.0
```

One caveat worth flagging: the inner gate's `1e-8 * rho_sc` threshold is *not*
reusable here. `rho_sc = max|rho_lm|` over the whole radial range is set by the
inner cusp, which for NFW is ~9 orders of magnitude larger than `rho_lm(r_max)`
— so `|rho_lm(r_max)| > 1e-8 * rho_sc` is false for every mode and would
disable the outer tail entirely. (I made exactly that mistake first; it removes
20% of `Phi_00` and up to 87% of the high-`l` modes at `r_max`, which is a
useful measure of how load-bearing the tail is.) If a relative threshold is
wanted, it should be scaled against something local to the outer boundary.

Happy to open a PR if useful.
